numpy/core/_type_aliases.py

```diff
@@ -100,8 +79,9 @@ def _bits_of(obj):
 def bitname(obj):
     """Return a bit-width name for a given type object"""
     bits = _bits_of(obj)
-    char = dtype(obj).kind
-    base = _kind_to_stem[char]
+    dt = dtype(obj)
+    char = dt.kind
+    base = _kind_name(dt)
```

Comment:
Failure here now raises `RuntimeError` rather than `KeyError`, which seems more useful to me.